service/matching/forwarder.go

```diff
@@ -136,12 +135,8 @@ func (fwdr *Forwarder) ForwardTask(ctx context.Context, task *internalTask) erro
 	var err error
 
 	// todo: Vet recomputing ScheduleToStart and rechecking expiry here
-	expiryGo, err := types.TimestampFromProto(task.event.Data.ExpiryTime)
-	if err != nil {
-		return err
-	}
-
-	newScheduleToStartTimeout := convert.Int32Ceil(time.Until(expiryGo).Seconds())
+	expiryGo := timestamp.TimeValue(task.event.Data.ExpiryTime)
+	newScheduleToStartTimeout := time.Until(expiryGo)
```

Comment:
This is implicit use of `time.Now`

Reply:
Will come back and add Until to Timestamp utils and replace functions across code base at one time